src/preview.cpp

```diff
@@ -807,9 +807,21 @@ namespace {
                     for (long i = 0; i < sizes.count(); i++) {
                         uint32_t offset = dataValue.toLong(i);
                         uint32_t size = sizes.toLong(i);
-                        enforce(Safe::add(idxBuf, size) < size_, kerCorruptedMetadata);
-                        if (size!=0 && Safe::add(offset, size) <= static_cast<uint32_t>(io.size()))
+
+                        // the size_ parameter is originally computed by summing all values inside sizes
+                        // see the constructor of LoaderTiff
+                        // But e.g in malicious files some of thes values could be negative
+                        // That's why we check again for each step here to really make sure we don't overstep
+                        //
+                        // if we are in the last iteration we check for equality since size_ = sum(sizes)
+                        enforce(Safe::add(idxBuf, size) < size_ ||
```

Comment:
Was #1829 caused by the enforce using `<` when it should have been `<=`?

```suggestion
                        enforce(Safe::add(idxBuf, size) <= size_ ||
```

Reply:
Yes, that's pretty much it. But I went for a bit stricter requirement and only allow `idxBuf+size == size_` on the last iteration of the for loop. 
That's when this case would happen on regular files.   

I'm not sure if it really makes a difference... want me to change it to just a simple `... <= size_` ?

Reply:
Thinking about it a bit I did like the simpler version better in the end so made the change to use `<=`

Reply:
Thinking about it a bit I did like the simpler version better in the end so made the change to use `<=`

Reply:
I have no objection to the extra check on the final iteration. I was just thinking about the edge case where the size is 0 on the final iteration.

Reply:
That's what I came to wonder as well and then ended up changing it :sweat_smile:   

I'm not sure if a size 0 is an actually valid entry, so I think the current solution is the safest